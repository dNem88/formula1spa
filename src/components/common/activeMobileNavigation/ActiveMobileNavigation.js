import React  from 'react';
import styles from './activeMobileNavigation.module.css'
import NavLink from '../NavLink/NavLink'
import navigation from '../../../utils/navigation/navigation'

function ActiveMobileNavigation({click, isLogged}) {
    const {guest, user} = navigation;
    return (
           <div onClick={click} className={styles['links-container']}>
               {isLogged ? user.map(x => {
                   return <NavLink key={x.href} {...x} classname={'mobile-link'}/>
               }) :
                    guest.map(x => {
                        return <NavLink key={x.href} {...x} classname={'mobile-link'}/>
                    })
               }
           </div>
    )
}

export default ActiveMobileNavigation