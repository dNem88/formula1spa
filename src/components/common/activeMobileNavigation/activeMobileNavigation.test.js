import React from 'react'
import renderer from 'react-test-renderer'
import ActiveMobileNavigation from './ActiveMobileNavigation'
import {
    MemoryRouter
} from 'react-router-dom'


const mockLogged = {
    click: () => {
        return 'mock'
    },
    isLogged: true
}
const mockGuest = {
     click: () => {
        return 'mock'
    },
    isLogged: false
}
describe('Testing MobileNavigation in Active State', () => {
    it('Render Link User is NOT Logged', () => {
        let snapshot = renderer.create(
            <MemoryRouter>
                <ActiveMobileNavigation {...mockGuest}/>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('div')
        expect(snapshot.props.className).toBe('links-container')
        expect(snapshot.children.length).toBe(9)
        expect(snapshot.children[7].type).toBe('a')
        expect(snapshot.children[7].props.href).toBe('/auth/login')
        expect(snapshot.children[7].children[0]).toBe('Sign in')
        expect(snapshot.children[8].type).toBe('a')
        expect(snapshot.children[8].props.href).toBe('/auth/register')
        expect(snapshot.children[8].children[0]).toBe('Register')
    })
     it('Render Link User is Logged In', () => {
        let snapshot = renderer.create(
            <MemoryRouter>
                <ActiveMobileNavigation {...mockLogged}/>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('div')
        expect(snapshot.props.className).toBe('links-container')
        expect(snapshot.children.length).toBe(8)
         expect(snapshot.children[7].type).toBe('a')
         expect(snapshot.children[7].props.href).toBe('/profile')
         expect(snapshot.children[7].children[0]).toBe('Profile')
    })
})