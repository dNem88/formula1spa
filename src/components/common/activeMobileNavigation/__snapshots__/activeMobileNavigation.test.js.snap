// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MobileNavigation in Active State Render Link User is Logged In 1`] = `
<div
  className="links-container"
  onClick={[Function]}
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    href="/news"
    onClick={[Function]}
  >
    News
  </a>
  <a
    href="/videos"
    onClick={[Function]}
  >
    Videos
  </a>
  <a
    href="/schedule"
    onClick={[Function]}
  >
    Schedule
  </a>
  <a
    href="/standings"
    onClick={[Function]}
  >
    Standings
  </a>
  <a
    href="/drivers"
    onClick={[Function]}
  >
    Drivers
  </a>
  <a
    href="/teams"
    onClick={[Function]}
  >
    Teams
  </a>
  <a
    href="/profile"
    onClick={[Function]}
  >
    Profile
  </a>
</div>
`;

exports[`Testing MobileNavigation in Active State Render Link User is NOT Logged 1`] = `
<div
  className="links-container"
  onClick={[Function]}
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    href="/news"
    onClick={[Function]}
  >
    News
  </a>
  <a
    href="/videos"
    onClick={[Function]}
  >
    Videos
  </a>
  <a
    href="/schedule"
    onClick={[Function]}
  >
    Schedule
  </a>
  <a
    href="/standings"
    onClick={[Function]}
  >
    Standings
  </a>
  <a
    href="/drivers"
    onClick={[Function]}
  >
    Drivers
  </a>
  <a
    href="/teams"
    onClick={[Function]}
  >
    Teams
  </a>
  <a
    href="/auth/login"
    onClick={[Function]}
  >
    Sign in
  </a>
  <a
    href="/auth/register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;
